compiler/luci-interpreter/src/core/Hook.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_CORE_HOOK_H
+#define LUCI_INTERPRETER_CORE_HOOK_H
+
+namespace luci_interpreter
+{
+
+// Used at execution stage to tell the interpreter that the runtime state has changed in some way.
+class Hook
```

Comment:
Please suggest a better name.

Reply:
How about Notifier?

This class notifies an event (change of runtime state according to your comment) to the interpreter.

Reply:
> How about Notifier?

I also considered this option. `Notifier` sounds too ambigous. I can't say that `Hook` is more specific though :)

Reply:
`EventNotifier` sounds good to me :)